projects/schematics/src/migrations/2_0/ssr/ssr.ts

```diff
@@ -0,0 +1,298 @@
+import {
+  apply,
+  chain, mergeWith, move,
+  Rule,
+  SchematicContext, SchematicsException, template,
+  Tree, UpdateRecorder, url,
+} from '@angular-devkit/schematics';
+import { getWorkspace } from '@schematics/angular/utility/workspace';
+import {strings} from "@angular-devkit/core";
+import * as ts from 'typescript';
+import {getDecoratorMetadata, getMetadataField} from "@schematics/angular/utility/ast-utils";
+import {
+  addPackageJsonDependency, NodeDependency,
+  NodeDependencyType,
+  removePackageJsonDependency
+} from "@schematics/angular/utility/dependencies";
+
+export function migrate(): Rule {
+  return async () => {
+    return chain([
+      backupExistingFiles(),
+      overwriteServerTsFile(),
+      modifyPackageJsonScripts(),
+      removeImportsInMainServerFile(),
+      removeMapLoaderModule(),
+      updateAngularJsonFile(),
+    ]);
+  };
+}
+
+
+export function backupExistingFiles() {
+  return (tree: Tree, context: SchematicContext) => {
+    context.logger.info('Backing up old server.ts and webpack.server.config.js files.');
+
+    const serverPath = '/server.ts';
+    const webpackPath = '/webpack.server.config.js';
+    const serverBuffer = tree.read(serverPath);
+    if (serverBuffer === null) {
+      throw new SchematicsException('Could not find server.ts file');
```

Comment:
Function to check whether SSR is used was implemented in the previous commit, but also good you pointed that out, because now this exception is unnecessary, thanks!